apps/rig_inbound_gateway/lib/rig_inbound_gateway_web/v1/longpolling_controller.ex

```diff
@@ -0,0 +1,96 @@
+defmodule RigInboundGatewayWeb.V1.LongpollingController do
+  @moduledoc """
+  Handling of Longpolling functionalitiy
+  """
+  require Logger
+  use Rig.Config, [:cors]
+
+  use RigInboundGatewayWeb, :controller
+
+  alias RigOutboundGateway
+  alias RigInboundGatewayWeb.Session
+  alias Rig.Connection
+
+  @doc false
+  def handle_connection(%{method: "GET"} = conn, _params) do
+    conn = conn |> fetch_cookies
+
+    conn.req_cookies["connection_token"] |> is_new_session? |> process_request(conn)
+  end
+
+  # -----
+  # Helpers
+  # -----
+
+  # validates if a connection_token was given. If yes, it validates if corresponding session processes is still alive, ignoring invalid/timed out cookies
+  defp is_new_session?(connection_token) do
+    case connection_token do
+      nil ->
+        true
+
+      connection_token ->
+        with {:ok, session_pid} <- Connection.Codec.deserialize(connection_token) do
+          !Process.alive?(session_pid)
+        else
+          _ -> false
+        end
+    end
+  end
+
+  # ---
+  defp process_request(is_new_session, conn)
+
+  # starts a new session
+  defp process_request(true, conn) do
```

Comment:
Looks like a lot of repetitive code, what do you think about updating it to `with` statement, such as in `apps/rig_inbound_gateway/lib/rig_inbound_gateway_web/v1/subscription_controller.ex`? Flow there is pretty similar.

Reply:
I had it that way, but kevin pointed out that doing it this way increases the readability of the main function so i changed it to that